engine/schema/src/com/cloud/network/dao/LoadBalancerDao.java

```diff
@@ -29,4 +30,6 @@
 
     List<LoadBalancerVO> listInTransitionStateByNetworkIdAndScheme(long networkId, Scheme scheme);
 
+    boolean isLoadBalancerRulesMappedToVmGuestIp(long instanceId, String instanceIp, long networkId, FirewallRule.Purpose purpose);
```

Comment:
@niteshsarda You don't need the purpose parameter here, as the implementation is specific for firewall rules. While setting the parameter with SearchBuilder the value can be set as  Purpose.LoadBalancing.